fetch.bs

```diff
@@ -2118,6 +2118,50 @@ clearly stipulates that <a>connections</a> are keyed on
      WebSocket saner -->
 
 
+<h3 id=http-cache-partitions>HTTP cache partitions</h3>
+
+<p>To
+<dfn lt="determine the HTTP cache partition|determining the HTTP cache partition">determine the HTTP cache partition</dfn>,
+given <var>request</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>topLevelOrigin</var> be null.
+
+ <li>
+  <p>If <var>request</var>'s <a for=request>reserved client</a> is non-null, then:
+
+  <ol>
+   <li><p>Set <var>topLevelOrigin</var> to <var>request</var>'s <a for=request>reserved client</a>'s
+   <a for="environment">top-level origin</a>.
+
+   <li>
+    <p>If <var>topLevelOrigin</var> is null, then set <var>topLevelOrigin</var> to
+    <var>request</var>'s <a for=request>reserved client</a>'s
+    <a for="environment">top-level creation URL</a>'s <a for=url>origin</a>.
+
+    <p class=note>This happens for top-level navigations only.
+  </ol>
```

Comment:
Can you remind me why we're doing this here, instead of having HTML ensure top-level origin is always non-null by setting it to the origin of the top-level creation URL?

Reply:
My thinking was that a null -> origin transition would be clearer than an origin -> origin transition. And it would also make it clearer here that we compute the origin directly from a URL rather than the response.

Reply:
Hmm, that feels like it kind of defeats the point of environment/ESO being an abstraction that other specs don't have to worth about the details of, but can just use and let HTML handle the management. But I don't feel too strongly.

Reply:
I think generally that's true and will remain true, but there's a couple places that will need to care about the distinction and networking is one of those. (Other places generally only get access to an environment settings object.)